packages/blade/src/components/ActionList/ActionListHeader.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import styled from 'styled-components';
+import { componentIds } from './componentIds';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { makeSize } from '~utils';
+import { Text } from '~components/Typography';
+
+const StyledActionListHeader = styled(Box)((props) => {
+  return {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    padding: `${makeSize(props.theme.spacing[3])} ${makeSize(props.theme.spacing[5])}`,
+    backgroundColor: props.theme.colors.brand.gray.a50.lowContrast,
+    margin: `-${makeSize(props.theme.spacing[3])} -${makeSize(props.theme.spacing[3])} ${makeSize(
+      props.theme.spacing[3],
+    )}`,
+  };
+});
+
+const ActionListHeader = (props: { title: string; leading?: React.ReactNode }): JSX.Element => {
```

Comment:
We can do this without a breaking change -

```jsx
<ActionListHeader>
  <TextInput />
</ActionListHeader>
```

Or a bit uglier version - 

```jsx
<ActionListHeader leading={<TextInput />} />
```

It can also just be default based on a `variant="filterable"` on Dropdown maybe? 

It seems rare that we will need a breaking change to support input.